File: Arrays/3610-find-x-sum-of-all-k-long-subarrays-i/find-x-sum-of-all-k-long-subarrays-i.py
class Solution:
    def helper(self, nums, n, x):
        count = defaultdict(int)

        for num in nums:
            count[num] += 1

        arr = [(key, value) for key, value in count.items()]
        arr.sort(key=lambda a: (a[1], a[0]), reverse=True)

        total = 0
        for i in range(min(len(arr), x)):
            total += arr[i][1] * arr[i][0]
        
        return total

    def findXSum(self, nums: List[int], k: int, x: int) -> List[int]:
        ans = []
        l = len(nums)

        for i in range(l - k + 1):
            ans.append(self.helper(nums[i:i+k], k, x))
        
        return ans
        

File: Arrays/3610-find-x-sum-of-all-k-long-subarrays-i/find-x-sum-of-all-k-long-subarrays-i.cpp
class Solution {
private:
    int helper(vector<int> nums, int n, int x) {
        unordered_map<int, int> count;

        for(int i = 0; i < n; i++) 
            count[nums[i]]++;

        vector<pair<int, int>> arr;
        for(auto& a : count)
            arr.push_back(pair<int,int>{a.first, a.second});
        
        sort(arr.begin(), arr.end(), [](auto &a, auto &b) {
            if(a.second == b.second) return a.first > b.first;
            return a.second > b.second;
        });

        int sum = 0;
        for(int i = 0; i < arr.size() && i < x; i++)
            sum += arr[i].second * arr[i].first;
        
        return sum;
    }

public:
    vector<int> findXSum(vector<int>& nums, int k, int x) {
        int n = nums.size();
        vector<int> ans;

        for(int i = 0; i <= n - k; i++) {
            vector<int> sub(nums.begin() + i, nums.begin() + i + k);
            ans.push_back(helper(sub, k, x));
        }

        return ans;
    }
};
